### PR TITLE
Radstorm harvesting

### LINKED ISSUE #29989: Radstorm Radiation is NOT harvestable.
Literally unplayable. How am I supposed to make the rare and mostly useless Radstorm Engine now. [consistency]

### PR DESCRIPTION
**Pull request by DeityLink**
Fixes #29989

:cl:
* rscadd: The radiations emitted by radstorms are now also harvested by radiation collectors and antimatter decelerators.

### CODE REVIEW

**Comment by Eneocho** [comment]
Can you make the radcells rads harvestable too?

**Comment by DeityLink** [comment]
no

**Comment by west3436** [comment]
what about radio transmitter radiation

**Comment by DeityLink** [comment]
> 
> 
> what about radio transmitter radiation

Honestly I don't think radcells or broadcasters emit nearly enough radition to be considered on par with a singularity or a supermatter, or even a cosmological event such as a radstorm. The effects of their radiations on humans are similar for gameplay reasons I guess, but given the energy cost of those machines/cells, the power generated would either be negligible, or lead to cheese by dumping a pile of infinite radcells on a single tile.

Something this low effort making actual engines obsolete doesn't have its place in spess imo. I might sound like the fun police here but I'm certain that it would have dire consequences for the fun in other areas.

**Comment by Eneocho** [comment]
> 
> 
> > what about radio transmitter radiation
> 
> Honestly I don't think radcells or broadcasters emit nearly enough radition to be considered on par with a singularity or a supermatter, or even a cosmological event such as a radstorm. The effects of their radiations on humans are similar for gameplay reasons I guess, but given the energy cost of those machines/cells, the power generated would either be negligible

I'd say broadcasters they emit the same amount of radiation as a barebones standard 3 laser shard, enough to cuck you fast (lord knows all the people that enter the DJ station, set up the broadcaster-juke combo without a radsuit and get cucked by rads in around 10 seconds) If the engi wants to do apply retardation and make a broadcaster engine, he should be able to.

>or lead to cheese by dumping a pile of infinite radcells on a single tile.

That assumes you have a constant source of EMPs to discharge the radcell, since radcells ONLY produce rads if they're charging (which isn't how RTGs work IRL, but whatever). Even assuming you had a constant EMP source, the radcells get damaged with each EMP, reducing the maximum charge they can hold/charge rate, to the point that if you actually managed to get 100 cells to "harvest power" from, you'd get cucked outta them rads after 3 or 4 EMP cycles.

And assuming all those problems are magically circumvented, you still need to make the 100 radcells in the first place, which needs both miners, scientists and a working engine to keep the station afloat while you made the retardcell engine.

>Something this low effort making actual engines obsolete doesn't have its place in spess imo.

I fail to see how would radcell retardation be low effort, you'd need a lot of radcells for the engine to be worth it (I.e: at least enough to charge 1 SMES), which means pestering RnD and mining, keeping the station powered while you make the thing, acquiring a constant EMP source, building and setting up the radcollectors, filling the plasma, etc.
And we DO have some low(er) effort engines in the game, an AME is arguably the simplest of the engines. 

To have some rad sources be harvestable but others not is extremely inconsistent, which is why there's the occasional radcells being harvestable request.

**Comment by DeityLink** [comment]
> I'd say broadcasters they emit the same amount of radiation as a barebones standard 3 laser shard

More reasons to **not** have the effects of radiation on mobs have the same ratio to energy no matter the source.

> And we DO have some low(er) effort engines in the game, an AME is arguably the simplest of the engines.

Ah yes, the AME, the engine you set up early on to stall the power outages until you have a real engine set up, and that requires an ever higher quantity of fuel if you want to get a decent output out of it. 

Radcells were not meant to become replacement for singulo, Radio Broadcasters were never meant to be an alternative to a supermatter shard (lol). Even code-wise there is no correlation between the damage on mob and the energy generated.

> To have some rad sources be harvestable but others not is extremely inconsistent, which is why there's the occasional radcells being harvestable request.

The solution is obvious, have them not irradiate people anymore, duh

(and in case my sarcasm wasn't obvious, my point was that inconsistency in some places is preferable, and not all feature requests are to be granted)

**Comment by Eneocho** [comment]
> 
> 
> > I'd say broadcasters they emit the same amount of radiation as a barebones standard 3 laser shard
> 
> More reasons to **not** have the effects of radiation on mobs have the same ratio to energy no matter the source.

Simple as reducing the rads emitted and making it harvestable.

>Ah yes, the AME, the engine you set up early on to stall the power outages until you have a real engine set up, and that requires an ever higher quantity of fuel if you want to get a decent output out of it.

We literally have the giant tank that has 10000 fuel AND the antimatter decelerator, both of which enable relatively longterm (roundwise) power (Around 1-1.5h, more than enough to do the autismo project, including engine retardation). Purchase 2 AME parts and a big fuel tank. You end with around 1.5MW if you max the cores and use all the parts (you can use around 70% of the capacity and get enough to charge the 3 SMES units). 

>Radio Broadcasters were never meant to be an alternative to a supermatter shard. Even code-wise there is no correlation between the damage on mob and the energy generated.

Since the broadcaster doesn't make harvestable power, I compared it with another rad metric, the damage to mob. Again, as simple as reducing the effect it has on radcollectors. Also consider that it's a powerhungry machine, enough to offset its own power production.

>The solution is obvious, have them not irradiate people anymore.

I actually tried the removing rads from radcells at one point, asking both in deadchat and on the c*der d*scord. I was convinced to keep it and ended up making the cells get damaged per EMP instead (See #29309). Radcells not being radioactive would arguably be more damaging to proper engines than making their rads harvestable, since you only need to swap a few cells and most areas become completely independent of power. 

I can concede on the radio broadcaster since it never made sense to me that its radioactive but tcomms isn't (don't be getting any ideas).

**Comment by DeityLink** [comment]
Or we could just keep radcells irradiating mobs, but their radiation not being harvestable.

Here I found something to satisfy immersions: Singulos, Supermatter (both capable of destroying matter at the nuclear level), and Radstorms (large scale cosmological events) emit **Neutron Radiation** that can only be blocked by the reinforced lining of maintenance hulls. As a matter of fact, rad collectors can harvest energy from a singulo or supermatter even if they don't have a straight line of sight to it.

Meanwhile, radio broadcasters and radcells emit "mere" **gamma (γ) radiation** which can be blocked by a single wall. And sure enough in the game, a wall is all that's needed between you and a broadcaster to not be affected (this is the case for most mob-related irradiation sources as well, such as exotic particle harvesters, or mobs affected with the radian syndrome).

Rad Collectors can only generate energy from the powerful neutron radiation, not from gamma rays.

In real life both types of rays are near equally deadly to the human body due to the way radiation affects cells.

TL;DR: Real life itself isn't always about consistency either.

**Comment by Eneocho** [comment]
Alright I'll concede, but we need the immulsion changes to reflect that. Though radcells would be alpha emitters rather than gamma emitters, since plutonium-238 (the most used RTG "fuel") is an alpha emitter (and would be perfectly shielded by the radcell's metal casing, so it wouldn't be radioactive. On the other hand, there's an argument to be made to make radcells more radioactive the more damaged it gets, like #29511 suggests). And the Radio Broadcaster using gamma rays instead of perfectly suitable radio waves (the station bounced _radios_ aren't radioactive, so they must be using radio waves bouncing on relays) doesn't make much sense. 
Also the whole thing about scrungulos making "radiation". Black body radiation is mostly thermic, not radioactive, so the radcollectors would actually be collecting heat, expanding the plasmagas via said heat and moving a local pump/generator using the expanded gas.

TL;DR: Immulsion changes needed if we follow your explanation, radcells should either not be radioactive (if we follow IRL logic) or increasingly radioactive as they get damaged, and radio broadcasters shouldn't be radioactive, which would need some remapping.
